deltachat-ffi/src/lib.rs

```diff
@@ -1262,19 +1273,20 @@ pub unsafe extern "C" fn dc_msg_new<'a>(
     assert!(!context.is_null());
     let context = &*context;
     if let Some(viewtype) = from_prim(viewtype) {
-        dc_msg::dc_msg_new(context, viewtype)
+        Box::into_raw(Box::new(dc_msg::dc_msg_new(context, viewtype)))
     } else {
         ptr::null_mut()
     }
 }
 
 #[no_mangle]
 pub unsafe extern "C" fn dc_msg_unref(msg: *mut dc_msg::dc_msg_t) {
-    dc_msg::dc_msg_unref(msg)
+    Box::from_raw(msg);
 }
 
 #[no_mangle]
 pub unsafe extern "C" fn dc_msg_empty(msg: *mut dc_msg::dc_msg_t) {
+    let msg = &mut *msg;
```

Comment:
I think you can avoid all those conversions from pointer to reference by doing
```
pub unsafe extern "C" fn dc_msg_empty(msg: &mut dc_msg::dc_msg_t) {
```

Reply:
Would `cbindgen` be happy?

Reply:
`cbindgen` is not used anymore: https://github.com/deltachat/deltachat-core-rust/commit/2acaf54f6905684f4b42cbd4a96cc6ea4db13cea

Reply:
but we want to bring it back, so lets keep thes interfaces as such for now